crypto/x509v3/v3_asid.c

```diff
@@ -730,7 +747,11 @@ static int asid_validate_path_internal(X509_STORE_CTX *ctx,
     } else {
         i = 0;
         x = sk_X509_value(chain, i);
-        OPENSSL_assert(x != NULL);
+        if (!ossl_assert(x != NULL)) {
```

Comment:
Just to double-clarify. "I'm with Rich" in sense that I'm in favour of *omitting* referred asserts instead of replacing them with ossl_assert. Well, to be utterly honest I for one wouldn't be actually against "soft" assert. But that's apparently just me. Anyway my rationale is following. Besides serving as kind of documentation purpose soft asserts can become useful for troubleshooting. Imagine report that says "it crashes". We are likely to ask reporter to examine core file. But chances also are that we will also ask to reproduce the problem with debug build. Because stack backtrace for optimized code might be less usable. But in debug build assert message might be sufficient to pinpoint the origin of the problem. Yes, one can argue that ossl_assert does it too in debug build. Well, but ossl_assert will avoid the crash, so that report would be "it doesn't work" instead of "it crashes", and suggestion to reproduce the problem with debug build wouldn't be first one to suggest. It would rather be "hmm, error message makes no sense, how could it happen", "have you missed this in manual", "try this", etc.